agent/agent_util.go

```diff
@@ -45,4 +46,32 @@ func getUsername() (string, error) {
 	} else {
 		return userInfo.Username, nil
 	}
-}
\ No newline at end of file
+}
+
+func getHostIPAddrs() ([]string, error) {
+    ifaces, err := net.Interfaces()
+    if err != nil {
+        return nil, err
+    }
+    var ipAddrs []string
+    for _, i := range ifaces {
+        addrs, err := i.Addrs()
+        if err != nil {
+            return nil, err
+        }
+        for _, addr := range addrs {
+            var ip net.IP
+            switch v := addr.(type) {
+            case *net.IPNet:
+                    ip = v.IP
+            case *net.IPAddr:
+                    ip = v.IP
+            }
+            ipv4 := ip.To4()
+            if ipv4 != nil && !ip.IsLoopback() {
+                ipAddrs = append(ipAddrs, ipv4.String())
+            }
+        }
+    }
+    return ipAddrs, nil
+}
```

Comment:
This looks pretty similar to `getLocalIPv4Addresses()` in proxy_receiver_http.go, do you think you can lift both out to reduce code repetition?
Possible untested solution: move `getLocalIPv4Addresses()` from proxy_receiver_http.go to proxy_util.go, update the function call in `getReachableUrls()` to just `GetLocalIPv4Addresses()`, then call `proxy.GetLocalIPv4Addresses()` in agent.go.

Reply:
Good idea. I didn't know about that code. I'll try and pull it out, probably add it to agent_util.go since its functionality would be used outside of proxy.